### PR TITLE
Display afk rewards dynamically

### LINKED ISSUE #362: Display AFK rewards dynamically
## Issue
Currently, the rewards icons are static in the front. But the rewards are variable, sometimes you'll receive some type of items and the next time it could be a different type of items. It would look bad if we showed all of the items with a "0" when you're not getting any, which is what's currently going on. This happens in the Claim AFK rewards popup in Overworld, and also in the Victory overlay after the Battle.

Screenshot 2024-04-17 at 1 43 26 PM

See also the image here for the Victory overlay: https://github.com/lambdaclass/champions_of_mirra/pull/357#pullrequestreview-2004381196

## Solution
Display items dynamically in Claim AFK rewards popup, Victory overlay and any other place where necessary.

### PR DESCRIPTION
**Pull request by tkz00**
Closes #362

## Motivation

When claiming afk rewards the currencies available are variable, some currencies will have 0 generation rate and we were showing them in the popup, this shouldn't happen.

## Summary of changes

Created an afk reward detail prefab and instantiated it in the popup to show which currencies are up for claim.

## How has this been tested?

Change which currencies your user idly generates, either via normal or artificial means, such as modifying the database or the backend directly, check the claim afk rewards claim popup in the Kaline tree reflects this by showing those currencies.

### CODE REVIEW

File: client/Assets/Scripts/AfkRewards/AfkReward.cs
Comment (ncontinanza) [inline client/Assets/Scripts/AfkRewards/AfkReward.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/AfkRewards/AfkRewardRate.cs
Comment (ncontinanza) [inline client/Assets/Scripts/AfkRewards/AfkRewardRate.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/Campaign/LevelData.cs
Comment (ncontinanza) [inline client/Assets/Scripts/Campaign/LevelData.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/Shared/GlobalUserData.cs
Comment (ncontinanza) [inline client/Assets/Scripts/Shared/GlobalUserData.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/Shared/HeaderManager.cs
Comment (ncontinanza) [inline client/Assets/Scripts/Shared/HeaderManager.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/Shared/UIReward.cs
Comment (ncontinanza) [inline client/Assets/Scripts/Shared/UIReward.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/Summon/Box.cs
Comment (ncontinanza) [inline client/Assets/Scripts/Summon/Box.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/UnitDetail/UIEquipmentListElement.cs
Comment (ncontinanza) [inline client/Assets/Scripts/UnitDetail/UIEquipmentListElement.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/UnitDetail/UnitDetail.cs
Comment (ncontinanza) [inline client/Assets/Scripts/UnitDetail/UnitDetail.cs:1]: Watch out for formatting here!

File: client/Assets/Scripts/User/User.cs
Comment (ncontinanza) [inline client/Assets/Scripts/User/User.cs:1]: Watch out for formatting here!

**Review by ncontinanza — CHANGES_REQUESTED** [review]

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Review by lotuuu — APPROVED** [review]
Works great! Doesn't even break if there are no rates available! Nice one

**Pull request merged**